File: LR2/app/src/main/java/com/mtpms/lr2/MainActivity.java
package com.mtpms.lr2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    public TextView result;
    public Button add;
    public EditText sumnum, procnum, daysnum;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        result = findViewById(R.id.result);
        add = findViewById(R.id.add);
        sumnum = findViewById(R.id.sumnum);
        procnum = findViewById(R.id.procnum);
        daysnum = findViewById(R.id.daysnum);

        add.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                double summa = Double.parseDouble(sumnum.getText().toString());
                double percents = Double.parseDouble(procnum.getText().toString());
                int days = Integer.parseInt(daysnum.getText().toString());
                double res = (summa * percents / 365 * days) / 100;
                result.setText(String.valueOf(res));
            }
        });
    }
}

File: LR15/LR15_1/app/src/main/java/com/mtpms/lr15_1/services/MainActivity.kt
package com.mtpms.lr15_1.services

import android.Manifest
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.content.pm.PackageManager
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.mtpms.lr15_1.stpms.services.R
import com.mtpms.lr15_1.services.services.MyLocationService
import kotlinx.android.synthetic.main.activity_main.*

private const val REQUEST_CODE: Int = 101
const val ACTION = "com.mtpms.lr15_1.services"

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        requestServicesWithPermissionCheck()
    }

    private fun requestServicesWithPermissionCheck() {

        val permissions = arrayListOf(
                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION),
                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION),
                ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
        )

        if (permissions.any { it != PackageManager.PERMISSION_GRANTED }) {

            requestPermissions(
                    arrayOf(
                            Manifest.permission.ACCESS_FINE_LOCATION,
                            Manifest.permission.ACCESS_COARSE_LOCATION,
                            Manifest.permission.INTERNET,
                            Manifest.permission.WRITE_EXTERNAL_STORAGE
                    ),
                    REQUEST_CODE
            )

        } else {
            requestLocation()
        }
    }

    private fun requestLocation() {
        button_start.setOnClickListener {
            val service = Intent(this, MyLocationService::class.java)
            ContextCompat.startForegroundService(this, service)
        }
        button_stop.setOnClickListener {
            val service = Intent(this, MyLocationService::class.java)
            stopService(service)
        }
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ) {

        when (requestCode) {
            REQUEST_CODE -> {
                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                    requestLocation()
                }
            }

        }
    }



    private val receiver = object: BroadcastReceiver() {
        override fun onReceive(p0: Context?, p1: Intent?) {
            Toast
                    .makeText(this@MainActivity, p1?.getStringExtra("success"), Toast.LENGTH_SHORT)
                    .show()
        }

    }

    private val filter = IntentFilter(ACTION)

    override fun onResume() {
        registerReceiver(receiver, filter)
        super.onResume()
    }

    override fun onPause() {
        unregisterReceiver(receiver)
        super.onPause()
    }
}